src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -247,11 +247,6 @@ create_distributed_table(PG_FUNCTION_ARGS)
 		shardCountIsStrict = true;
 	}
 
-	EnsureCitusTableCanBeCreated(relationId);
-
-	/* enable create_distributed_table on an empty node */
-	InsertCoordinatorIfClusterEmpty();
-
 	/*
 	 * Lock target relation with an exclusive lock - there's no way to make
 	 * sense of this table until we've committed, and we don't want multiple
```

Comment:
It might be the case that these operations were intentionally placed before taking an exclusive lock.

For InsertCoordinatorIfClusterEmpty, I'm not sure that's really true.

EnsureCitusTableCanBeCreated checks EnsureTableOwner, so random users would not be able to take a lock.

Can the lock move down as well?

Reply:
let me think about these little more

Reply:
> Can the lock move down as well?

I think that makes sense, let me fix the tests and ask for a review.

Reply:
Moved relation_open / locking logic down to CreateCitusTable too.